chopper.py

```diff
@@ -191,33 +227,35 @@ def clean_filename(string: str) -> str:
     return re.sub(r"\W", "_", string)
 
 
-def split_by_columns(filepath: Path, col_list: list[str], config: Config) -> list[Path]:
+def split_by_columns(
+    filepath: Path, columns: list[str], config: ChopperNamespace
+) -> list[Path]:
     """Create one file per unique combination of values in the specified columns in
     the input file.
 
         Args:
             filepath (Path): The path of the file to split.
-            col_list (list): List of columns to split by. Must match values in header row exactly.
-            config (Config): Config object created from CLI arguments.
+            columns (list[str]): List of columns to split by. Must match values in header row exactly.
+            config (argpase.Namespace): Namespace object created from CLI arguments.
 
         Returns:
             list[Path]: List of split intermediate files.
     """
 
-    with open(filepath, "r", encoding=config.encoding) as f:
-        reader = csv.DictReader(f, delimiter=config.delimiter)
+    with filepath.open("r", encoding=config.encoding, newline="") as fin_obj:
+        reader = csv.DictReader(fin_obj, delimiter=config.delimiter)
```

Comment:
Unfortunately, I haven't yet figured out how to get DictReader working with mmap because it expects strings not bytes, so this remains relatively slow.